src/CouponCode.php

```diff
@@ -233,22 +233,24 @@ protected function _normalize($string, array $options = []) {
 		return $string;
 	}
 
-	/**
-	 * Generates a cryptographically secure sequence of bytes.
-	 *
-	 * @param integer $bytes Number of bytes to return.
-	 * @return string
-	 */
+    /**
+     * Generates a cryptographically secure sequence of bytes.
+     *
+     * @param integer $bytes Number of bytes to return.
+     * @return string
+     * @throws Exception
+     */
 	protected function _random($bytes) {
-		if (is_readable('/dev/urandom')) {
-			$stream = fopen('/dev/urandom', 'rb');
-			$result = fread($stream, $bytes);
+        //if (is_readable('/dev/urandom')) {
+        if ($fh = @fopen('/dev/urandom', 'rb')) {
+            $stream = fopen('/dev/urandom', 'rb');
+            $result = fread($stream, $bytes);
 
-			fclose($stream);
-			return $result;
-		}
+            fclose($stream);
+            return $result;
+        }
 		if (function_exists('mcrypt_create_iv')) {
-			return mcrypt_create_iv($bytes, MCRYPT_DEV_RANDOM);
+			return random_bytes($bytes);
```

Comment:
This line will not be reachable unless the mcrypt extension is installed, which is not what you want. How about adding a similar block as a first possible source of randomness in our `_random()` method here? That'd come before the check for using `/dev/urandom`. Before being able to use `random_bytes()` check if the function is present, older PHP versions may not have it and we still want to support those.